src/test/java/com/williamfiset/algorithms/datastructures/balancedtree/RedBlackTreeTest.java

```diff
@@ -246,51 +246,72 @@ public void testRandomizedValueInsertionsAgainstTreeSet() {
     }
   }
 
-  /*
-    @Test
-    public void randomRemoveTests() {
-      TreeSet<Integer> ts = new TreeSet<>();
-      for (int i = 0; i < TEST_SZ; i++) {
-
-        int size = i;
-        List <Integer> lst = genRandList(size);
-        for (Integer value : lst) {
-          tree.insert(value);
-          ts.add(value);
-        }
-        Collections.shuffle(lst);
-
-        // Remove all the elements we just placed in the tree.
-        for (int j = 0; j < size; j++) {
-
-          Integer value = lst.get(j);
-          assertEquals(ts.remove(value), tree.remove(value));
-          assertFalse(tree.contains(value));
-          assertEquals(size - j - 1, tree.size());
-        }
-        assertTrue(tree.isEmpty());
-      }
-    }
+  @Test
+  public void testRemoval() {
+    tree.insert(5);
+    tree.insert(7);
+    tree.insert(9);
+
+    tree.delete(5);
+    assertFalse(tree.contains(5));
+
+    tree.delete(7);
+    assertFalse(tree.contains(7));
 
-    @Test
-    public void testTreeHeight() {
-      for (int n = 1; n <= TEST_SZ; n++) {
+    tree.delete(9);
+    assertFalse(tree.contains(9));
+  }
+
+  @Test
+  public void testNullRemoval() {
+    assertFalse(tree.delete(null));
+  }
+
+  @Test
+  public void testNumberDoesntExist() {
+    assertFalse(tree.delete(0));
+  }
 
-        tree.insert(randValue());
-        int height = tree.height();
+  @Test
+  public void randomRemoveTests() {
+    TreeSet<Integer> ts = new TreeSet<>();
+    for (int i = 0; i < TEST_SZ; i++) {
 
-        // Get an upper bound on what the maximum height of
-        // an AVL tree should be. Values were taken from:
-        // https://en.wikipedia.org/wiki/AVL_tree#Comparison_to_other_structures
-        double c = 1.441;
-        double b = -0.329;
-        double upperBound = c*(Math.log(n+2.0)/Math.log(2)) + b;
+      List<Integer> lst = genRandList(i);
+      for (Integer value : lst) {
+        tree.insert(value);
+        ts.add(value);
+      }
+      Collections.shuffle(lst);
 
-        assertTrue(height < upperBound);
+      for (int j = 0; j < i; j++) {
 
+        Integer value = lst.get(j);
+        assertEquals(ts.remove(value), tree.delete(value));
+        assertFalse(tree.contains(value));
+        assertEquals(i - j - 1, tree.size());
       }
+      assertTrue(tree.isEmpty());
+    }
+  }
+
+  @Test
+  public void testTreeHeight() {
```

Comment:
Don't feel the need to include the tree height test and method if it's causing a headache. This is more important for AVL trees than for RB trees. Feel free to remove if you want.

Reply:
Oh thanks! But it wasn't that, there were some bugs in my actual delete method, and since I made some stuff really messy, I'm thinking of redoing it. It won't be much though, since all I need to do is copy and paste my source code.